openupgrade_scripts/scripts/point_of_sale/18.0.1.0.2/post-migration.py

```diff
@@ -0,0 +1,119 @@
+
+
+def fill_pos_order_reversed_pos_order_id(env):
+    openupgrade.logged_query(
+        env.cr,
+        """
+        UPDATE account_move am2
+        SET reversed_pos_order_id = po.id
+        FROM pos_order po
+        JOIN pos_session ps ON po.session_id = ps.id AND ps.state = 'closed'
+        JOIN pos_config pc ON ps.config_id = pc.id
+        JOIN account_journal aj ON pc.journal_id = aj.id
+        JOIN account_move am ON po.account_move = am.id
+            AND am.state = 'posted' AND am.journal_id = pc.invoice_journal_id
+            AND am.invoice_origin = po.name
+        JOIN account_move session_move ON session_move.id = ps.move_id
+        WHERE po.partner_id IS NOT NULL AND
+            am2.journal_id = pc.journal_id
+            -- we find correct move by looking into the ref
+            AND am2.ref like CONCAT('%', session_move.name, '%')
+            AND am2.ref like CONCAT('% ', po.name, '%')
+            AND am2.ref like CONCAT('%', ps.name, '%')""",
+    )
+
+
+def fill_pos_config_customer_display_type(env):
+    openupgrade.logged_query(
+        env.cr,
+        """
+        UPDATE pos_config pc
+        SET customer_display_type = CASE
+            WHEN pc.iface_customer_facing_display_via_proxy THEN 'proxy'
+            WHEN pc.iface_customer_facing_display_local THEN 'local'
+            ELSE 'none' END""",
+    )
+
+
+def update_pos_config_show_images(env):
+    show_product_images = (
+        env["ir.config_parameter"]
+        .sudo()
+        .get_param("point_of_sale.show_product_images", "yes")
+    )
+    if show_product_images == "no":
+        openupgrade.logged_query(
+            env.cr,
+            """
+            UPDATE pos_config pc
+            SET show_product_images = FALSE
+            """,
+        )
+    show_category_images = (
+        env["ir.config_parameter"]
+        .sudo()
+        .get_param("point_of_sale.show_category_images", "yes")
+    )
+    if show_category_images == "no":
+        openupgrade.logged_query(
+            env.cr,
+            """
+            UPDATE pos_config pc
+            SET show_category_images = FALSE
+            """,
+        )
+
+
+def fill_pos_uuid(env):
+    pos_orders = env["por.order"].search([])
+    for pos_order in pos_orders:
+        pos_order.uuid = str(uuid4())
+    pos_payments = env["por.payment"].search([])
+    for pos_payment in pos_payments:
+        pos_payment.uuid = str(uuid4())
+
+
+def update_res_company_point_of_sale_ticket_portal_url_display_mode(env):
+    # in v17, url was always shown
+    openupgrade.logged_query(
+        env.cr,
+        """
+        UPDATE res_company rc
+        SET point_of_sale_ticket_portal_url_display_mode = CASE
+            WHEN rc.point_of_sale_use_ticket_qr_code THEN 'qr_code_and_url'
+            ELSE 'url' END""",
+    )
+
+
+@openupgrade.migrate()
+def migrate(env, version):
+    fill_pos_order_reversed_pos_order_id(env)
+    fill_pos_config_customer_display_type(env)
```

Comment:
```suggestion
    fill_pos_config_customer_display_type(env)
    fill_pos_uuid(env)
```

the dropping coverage could have tipped you off about this one